test/python/quantum_info/test_sparse_observable.py

```diff
@@ -155,6 +155,17 @@ def test_default_constructor_copy(self):
         with self.assertRaisesRegex(ValueError, "explicitly given 'num_qubits'"):
             SparseObservable(base, num_qubits=base.num_qubits + 1)
 
+    def test_default_constructor_term(self):
+        expected = SparseObservable.from_list([("IIZXII+-", 2j)])
+        self.assertEqual(SparseObservable(expected[0]), expected)
+
+    def test_default_constructor_term_iterable(self):
+        expected = SparseObservable.from_list([("IIZXII+-", 2j), ("rlIIIIII", 0.5)])
+        terms = [expected[0], expected[1]]
+        self.assertEqual(SparseObservable(list(terms)), expected)
+        self.assertEqual(SparseObservable(tuple(terms)), expected)
+        self.assertEqual(SparseObservable(term for term in terms), expected)
```

Comment:
Is this the same as this?
```suggestion
        self.assertEqual(SparseObservable(iter(terms)), expected)
```

Reply:
Yeah, it'll end up being effectively the same test.  Do you find the intent of your suggestion clearer?  I'll change it if so.

Reply:
No that's perfectly fine, just to clarify 🙂 (but using `iter` would look aesthetically more pleasing 😛)